src/pages/visualization-view/visualization-view.js

```diff
@@ -37,7 +37,7 @@ const VisualizationView = (props) => {
   const startIndex = idFromUrl ? 1 : 0
   const [index, setIndex] = useState(startIndex)
 
-  React.useEffect(() => { reset() }, [])
+  React.useEffect(() => { return function cleanup() { reset() }}, [])
```

Comment:
Does it make any sense to still reset on mount as well? Just in case some other page in the app also has set the visualization or query in the store?

Reply:
Also, did you look to see if there's any sensible way for us to test this behavior?

Reply:
Yeah you're right

Reply:
And in the process of doing so